### ISSUE TITLE
Incorrect `react` peer dependency version installed by default

### ISSUE THREAD

**Issue by wSedlacek** [nrwl/nx-react-native#56]
The following error happens when attempting to use the default configurations with NPM v7 (after correcting the incorrect @nrwl/react version #55)

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: foo@0.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR!   peer react@"*" from @testing-library/jest-native@3.4.3
npm ERR!   node_modules/@testing-library/jest-native
npm ERR!     dev @testing-library/jest-native@"3.4.3" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"16.13.1" from react-native@0.63.4
npm ERR! node_modules/react-native
npm ERR!   react-native@"0.63.4" from the root project
npm ERR!   peer react-native@"*" from @testing-library/jest-native@3.4.3
npm ERR!   node_modules/@testing-library/jest-native
npm ERR!     dev @testing-library/jest-native@"3.4.3" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```

**Comment by xiongemi** [comment]
create a pr to fix this: https://github.com/nrwl/nx-react-native/pull/69

**Comment by xiongemi** [comment]
it might be possible fixed in pr #68 for release 12.3.6. Now after this or, the @nrwl/workspace and @nrwl/react should be using version 12.3.6.
